File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.js
/**
 * @param {string[][]} items
 * @param {string} ruleKey
 * @param {string} ruleValue
 * @return {number}
 */
var countMatches = function(items, ruleKey, ruleValue) {
    let count = 0;
    let j;
    if(ruleKey === "color"){
            j = 1;
        }else if(ruleKey === "type"){
            j = 0;
        }else{
            j = 2;
        }
    for(let i=0; i<items.length; i++){
        if(items[i][j] === ruleValue){
            count += 1;
        }
    }
    return count;
};

File: 1773-count-items-matching-a-rule/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
        int count = 0;
        for(int i=0; i<items.size(); i++){
            System.out.println(items.get(i));
            // System.out.println(ruleKey+" "+ruleValue);
            if(ruleKey.equals("type")){
                if(ruleValue.equals(items.get(i).get(0))){
                    count += 1;
                }
            }else if(ruleKey.equals("color")){
                // System.out.println(ruleValue+" "+items.get(i).get(1));
                if(ruleValue.equals(items.get(i).get(1))){
                    count += 1;
                }
            }else if(ruleKey.equals("name")){
                if(ruleValue.equals(items.get(i).get(2))){
                    count += 1;
                }
            }
        }
        return count;
    }
}